system/lib/wasmfs/wasmfs.cpp

```diff
@@ -82,13 +87,21 @@ __wasi_errno_t __wasi_fd_write(__wasi_fd_t fd,
 
   auto lockedFile = file->locked();
 
-  off_t offset = lockedOpenFile.position();
+  off_t currOffset =
+    setOffset == OffsetHandling::overwrite ? lockedOpenFile.position() : offset;
+  off_t oldOffset = currOffset;
+  auto assignOffset{[&] {
+    *nwritten = currOffset - oldOffset;
+    if (setOffset == OffsetHandling::overwrite) {
+      lockedOpenFile.position() = currOffset;
+    }
+  }};
```

Comment:
This is how we typically write this:
```suggestion
  auto assignOffset = [&] {
    *nwritten = currOffset - oldOffset;
    if (setOffset == OffsetHandling::overwrite) {
      lockedOpenFile.position() = currOffset;
    }
  };
```
Also, since this also updates `*nwritten` and only conditionally updates the offset, how about we call this `finish`?